src/olympia/blocklist/models.py

```diff
@@ -256,49 +257,65 @@ def can_user_signoff(self, signoff_user):
             self.updated_by != signoff_user)
         return not require_different_users or different_users
 
-    def update_if_signoff_not_needed(self):
+    def all_adu_safe(self):
         threshold = settings.DUAL_SIGNOFF_AVERAGE_DAILY_USERS_THRESHOLD
 
-        def any_unsafe():
-            return any(
-                (lambda du: du > threshold or du == 0)(
-                    block['average_daily_users'])
-                for block in self.to_block)
+        return all(
+            (lambda du: du <= threshold and du)(block['average_daily_users'])
+            for block in self.to_block)
+
+    def has_version_changes(self):
+        block_ids = [block['id'] for block in self.to_block]
+
+        has_new_blocks = any(not id_ for id_ in block_ids)
+        blocks_with_version_changes_qs = Block.objects.filter(
+            id__in=block_ids).exclude(
+                min_version=self.min_version, max_version=self.max_version)
 
-        if self.signoff_state == self.SIGNOFF_PENDING and not any_unsafe():
-            self.update(signoff_state=self.SIGNOFF_NOTNEEDED)
+        return has_new_blocks or blocks_with_version_changes_qs.exists()
+
+    def update_if_signoff_not_needed(self):
+        is_pending = self.signoff_state == self.SIGNOFF_PENDING
+        add_action = self.action == self.ACTION_ADDCHANGE
+        if (
+            (is_pending and self.all_adu_safe()) or
+            (is_pending and add_action and not self.has_version_changes())
```

Comment:
When would a block have `has_version_changes() == True` but `self.action != self.ACTION_ADDCHANGE`? I.e., which use cases are we excluding with this rule?

Reply:
Only with `ACTION_ADDCHANGE` are the versions updated - with `ACTION_DELETE` the version properties aren't used (though do exist on the model instance) so we shouldn't consider if the values are different in any Blocks vs. this submission.

tbh, conceptually having a separate model for `ACTION_DELETE` may be clearer but model inheritance in django is tricky and can cause more problems than it solves.